src/api/app/services/workflows/scm_event_subscription_creator.rb

```diff
@@ -19,11 +29,24 @@ def call
                                              channel: 'scm',
                                              enabled: true,
                                              token: @token,
-                                             package: @package).tap do |subscription|
+                                             package: package).tap do |subscription|
           # Set payload and workflow_run regardless of whether the subscription already existed or not
           subscription.update!(workflow_run: @workflow_run, payload: @scm_webhook.payload)
         end
       end
     end
+
+    def create_or_update_subscriptions_for_request(bs_request:)
+      EventSubscription.find_or_create_by!(eventtype: 'Event::RequestStatechange',
+                                           receiver_role: 'reader', # We pass a valid value, but we don't need this.
+                                           user: @token.executor,
+                                           channel: 'scm',
+                                           enabled: true,
+                                           token: @token,
+                                           workflow_run: @workflow_run,
```

Comment:
This will never find, only create. We just fixed this bug in the method above: #14690

Reply:
Adapted